src/wizard/interactive_build.jl

```diff
@@ -184,6 +184,60 @@ function diff_srcdir(state::WizardState, prefix::Prefix, ur::Runner)
     return false
 end
 
+function bb_add(client, state::WizardState, prefix::Prefix, platform::Platform, jll::AbstractString)
+    if any(dep->getpkg(dep).name == jll, state.dependencies)
+        println(client, "ERROR: Package was already added")
+        return
+    end
+    new_dep = Dependency(jll)
+    try
+        # This will redo some work, but that may be ok
+        setup_dependencies(prefix, getpkg.([state.dependencies; new_dep]), platform)
+        push!(state.dependencies, new_dep)
+    catch e
+        showerror(client, e)
+    end
+end
+
+function setup_bb_service(state::WizardState, prefix, platform)
+    fpath = joinpath(prefix, "metadir", "bb_service")
+    server = listen(fpath)
```

Comment:
What OSes do we expect this to work on? I know Linux is the first class citizen, but in my testing it doesn't look like this works on MacOS or Windows.  Perhaps we should use a TCP port instead.

Reply:
It works on all OSes on the julia side. There's a question of how to forward it to the runner. Not sure docker can forward named pipes (it's not hard, but it is a bit of code). I would prefer this approach where possible, since TCP sockets kinda need to be authenticated to not open up a security issue. I was hoping to punt on that for now.

Reply:
> It works on all OSes on the julia side.

What am I doing wrong?

```
$ cat listen.jl 
using Sockets
listen("./foo")

$ julia listen.jl
ERROR: LoadError: ArgumentError: could not listen on path ./foo
Stacktrace:
 [1] listen(::String) at /Users/sabae/tmp/julia-build/julia-release-1.4/usr/share/julia/stdlib/v1.4/Sockets/src/PipeServer.jl:76
 [2] top-level scope at /Users/sabae/listen.jl:2
 [3] include(::Module, ::String) at ./Base.jl:377
 [4] exec_options(::Base.JLOptions) at ./client.jl:288
 [5] _start() at ./client.jl:484
in expression starting at /Users/sabae/listen.jl:2
```

Reply:
It works on Linux, BTW.

Reply:
Do you already have a "foo" in your current directory?

Reply:
Ah, you're right, that was the problem on MacOS.  But on Windows, I still get the same error:

```
julia> listen("foo")
ERROR: ArgumentError: could not listen on path foo
Stacktrace:
 [1] listen(::String) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.3\Sockets\src\PipeServer.jl:76
[2] top-level scope at REPL[8]:1

julia>

Administrator@win64bot3 ~
$ ls -la foo
ls: cannot access 'foo': No such file or directory
```

Reply:
> TCP sockets kinda need to be authenticated to not open up a security issue

Binding to `localhost` on a randomized port (provided to the client through an environment variable) would address most of the security concerns, I think.

Reply:
Windows is complicated. You need to use the pipe namespace. I guess that means we should make this unix only for now.

Reply:
> Binding to localhost on a randomized port (provided to the client through an environment variable) would address most of the security concerns, I think.

Not on a shared machine. I can scan all of localhost in about 50ms. We had a CVE on julia because we used to do that.